src/TreeHouse/EventStore/Upcasting/SimpleUpcasterChain.php

```diff
@@ -22,17 +22,33 @@ public function registerUpcaster(UpcasterInterface $upcaster)
     /**
      * Upcasts via a chain of upcasters.
      *
-     * @inheritdoc
+     * @return array|SerializedEvent[]
      */
     public function upcast(SerializedEvent $event, UpcastingContext $context)
     {
+        $result = [];
+        $events = [$event];
+
         foreach ($this->upcasters as $upcaster) {
-            if ($upcaster->supports($event)) {
-                $event = $upcaster->upcast($event, $context);
+            $result = [];
+
+            foreach ($events as $event) {
+                if ($upcaster->supports($event)) {
+                    $upcasted = $upcaster->upcast($event, $context);
+
+                    // TODO: deprecate support of non array values by upcasters
+                    if (!is_array($upcasted)) {
+                        $upcasted = [$upcasted];
```

Comment:
Please see if we can trigger a deprecation warning here, so that we can phase this out in the next major release.